exec/cnex/lib/net.c

```diff
@@ -355,7 +364,7 @@ void net_socket_select(TExecutor *exec)
     }
 
     for (size_t i = 0; i < read->array->size; ) {
-        SOCKET *ps = check_socket(exec, read->array->data[i].object);
+        SOCKET *ps = check_socket(exec, ((Array*)read->array->data[i].array)->data[0].object);
```

Comment:
I am still pretty sure that you do not need these casts. If you don't need them, remove them.

Reply:
Fixed.